packages/amazonq/test/unit/amazonqFeatureDev/util/files.test.ts

```diff
@@ -56,6 +68,37 @@ describe('file utils', () => {
             assertTelemetry('amazonq_bundleExtensionIgnored', { filenameExt: 'mp4', count: 1 })
         })
 
+        it('should ignore devfile.yaml when setting is disabled', async function () {
+            const workspace = await getWorkspaceWithDevFile()
+            sinon.stub(CodeWhispererSettings.instance, 'getDevCommandWorkspaceConfigurations').returns({})
+
+            const telemetry = new TelemetryHelper()
+            const result = await prepareRepoData([workspace.uri.fsPath], [workspace], telemetry, {
+                record: () => {},
+            } as unknown as Span<AmazonqCreateUpload>)
+            assert.strictEqual(Buffer.isBuffer(result.zipFileBuffer), true)
+            // checksum is not the same across different test executions because some unique random folder names are generated
+            assert.strictEqual(result.zipFileChecksum.length, 44)
+            assert.strictEqual(telemetry.repositorySize, 12)
+            sinon.reset()
+        })
+
+        it('should include devfile.yaml when setting is enabled', async function () {
+            const workspace = await getWorkspaceWithDevFile()
```

Comment:
One final thing, does it make sense to do:
```
function testDevfile(settings: {
     workspaceConfiguration: {} | boolean,
     repositorySize: number
}) {
    const workspace = await getWorkspaceWithDevFile()
    sinon.stub(CodeWhispererSettings.instance, 'getDevCommandWorkspaceConfigurations').returns({
        [workspace.uri.fsPath]: settings.workspaceConfiguration,
    })

    const telemetry = new TelemetryHelper()
    const result = await prepareRepoData([workspace.uri.fsPath], [workspace], telemetry, {
        record: () => {},
    } as unknown as Span<AmazonqCreateUpload>)
    assert.strictEqual(Buffer.isBuffer(result.zipFileBuffer), true)
    // checksum is not the same across different test executions because some unique random folder names are generated
    assert.strictEqual(result.zipFileChecksum.length, 44)
    assert.strictEqual(telemetry.repositorySize, settings.repositorySize)
}
```

```
it('should ignore devfile.yaml when setting is disabled', async function() {
     testDevfile({ workspaceConfiguration: {}, repositorySize: 12 })
}

it('should include devfile.yaml when setting is enabled', async function() {
     testDevfile({ workspaceConfiguration: true, repositorySize: 16 })
}
```

duplicate code now fails ci, I think we can probably de-dup some of this stuff

Reply:
if you want I can re-write the entire unit test file to avoid duplication completely, there are other areas that can be improved in terms of duplication as well